src/osd/PGBackend.cc

```diff
@@ -157,8 +157,6 @@ void PGBackend::handle_recovery_delete(OpRequestRef op)
     [=](int r) {
       if (r != -EAGAIN) {
 	get_parent()->send_message_osd_cluster(reply, conn.get());
-      } else {
-	reply->put();
```

Comment:
the regression was introduced by https://github.com/ceph/ceph/commit/d4a71c3c5369c7a881822a6eb7624e3e80cc53de

Reply:
you can add this information to the commit message